R/CreatePackage.R

```diff
@@ -1,58 +1,26 @@
-#' if strPackageName is provided, is not blank and bCreateProjectSubdirectory == TRUE then a folder named  strPackageName is created in the strProjectDirectory directory.}
-#' @param bCreateProjectSubdirectory {If bCreateProjectSubdirectory then then a subdirectory for the project is created in strProjectDirectory.  }
+#' @description {Create an R package via usethis::create_package. Optionally include a sample functiona and a testthat unit test.   }
+#' @param strDirectory {The directory where the project should be created.  If this parameter is left blank then the current working directory will be used.}
+#' @param strName {The name of the package folder.}
+#' @param bSampleFiles {copy sample files? Default: TRUE}
 #' @export
-CreatePackage <- function( strProjectDirectory, strPackageName, strAuthors = NULL, bCreateProjectSubdirectory = TRUE )
-{
-    # Set the template directory to copy from
-    strProjectDirectory         <- gsub( "\\\\", "/", strProjectDirectory )
-    strTemplateDirectory        <- GetTemplateDirectory()
-    strCalculationsDirectory    <- paste( strTemplateDirectory, "/CalculationPackage", sep="" )
-
-    strDestDirectory <- CreateProjectDirectory( strProjectDirectory, strPackageName, bCreateProjectSubdirectory )
-
-
-    #strDestDirectory            <- paste( strProjectDirectory, "/", strPackageName, sep="" )
-
-    strRet        <- CopyFiles( strCalculationsDirectory, strDestDirectory )
-    strCalcPkgRet <- UpdateCalculationPackageName( strProjectDirectory, strPackageName )
-    strUpdateAuthor <- UpdateAuthors(
-        paste( strProjectDirectory, "/", strShinyAppName, sep="" ),
-        strAuthors
-    )
-
-    strRet <- paste( c("Creating Calculation Package...", strRet, strCalcPkgRet, strUpdateAuthor), collapse="\n" )
-    return(strRet)
-}
 
-
-UpdateCalculationPackageName <- function(  strProjectDirectory, strPackageName )
+CreatePackage <- function(
+    strDirectory=getwd(),
+    strName="newPackage",
+    strAuthors = "",
+    bSampleFiles=TRUE)
 {
-
-    # Step 1: Rename project file ####
-    strPackageDir <- paste( strProjectDirectory, "/", strPackageName, sep = "" )
-    strRProjName  <- paste( strPackageDir, "/CalculationPackage.Rproj", sep="" )
-    if( file.exists( strRProjName ) )
-    {
-        strNewRProjName <- paste(strPackageDir, "/", strPackageName,".Rproj", sep = "" )
-
-        file.rename( strRProjName, strNewRProjName )
-        strRProjName  <- strNewRProjName
+    strPackagePath<-paste0(strDirectory,"/",strName)
+    opts <- list()
+    if(strAuthors != "") opts[`Authors@R`]=strAuthors
+    usethis::create_package(path=strPackagePath, fields=opts, check_name=FALSE, open=FALSE)
+    usethis::with_project(path=strPackagePath,usethis::use_testthat())
+
+    if(bSampleFiles){
+        strSrcDir<-paste0(GetTemplateDirectory(),"/Pkg")
+        strDestDir<- paste0(strPackagePath)
+        CopyFiles(strSrcDir, strDestDir)
     }
```

Comment:
It seems the AUTHOR_NAME tag is not created in the file, so the future rendering for that part is missing?

Reply:
Right - I've converted this to use the `fields` option in usethis::create_package to fill in the description file instead.